app/src/main/java/com/github/se/orator/ui/profile/PreviousRecordingsFeedback.kt

```diff
@@ -49,68 +49,77 @@ fun PreviousRecordingsFeedbackScreen(
-    // Call necessary methods or logic when the screen is opened
-    speakingViewModel.getTranscript(audioFile)
-    // Any other initialization logic
-    Log.d("prompts are: ", prompts?.get("targetPosition") ?: "Default Value")
-    viewModel.resetResponse()
   }
 
-  val response by viewModel.response.collectAsState("")
-
-  LaunchedEffect(response) { Log.d("gpt said: ", "gpt said: $response") }
-
-  if (offlineAnalysisData != null) {
-    viewModel.offlineRequest(
-        offlineAnalysisData!!.transcription.removePrefix("You said:").trim(),
-        prompts?.get("targetCompany") ?: "Apple",
-        prompts?.get("jobPosition") ?: "engineer")
-    Log.d("testing offline chat view model", "the gpt model offline value response is $response")
-    // Text(text = "What you said: ${what_has_been_said.value}")
-    Text(text = "Interviewer's response: $response", color = MaterialTheme.colorScheme.primary)
-    Log.d("d", "Hello! This is has been said: ${offlineAnalysisData!!.transcription}")
+  // if there isn't already an interviewer response: transcribe text + request a gpt prompt
+  if (fileData == "Loading interviewer response..." || fileData.isNullOrEmpty()) {
+    Log.d("in pre ", "file data is null! $fileData")
+    LaunchedEffect(speakingViewModel.isTranscribing.value) {
+      if (!speakingViewModel.isTranscribing.value)
+          speakingViewModel.getTranscriptAndGetGPTResponse(
+              audioFile, prompts, viewModel, context, offlinePromptsFunctions)
+    }
   }
-  // prompts?.get("targetPosition") ?: "Default Value"
-  // val jobPosition = prompts?.get("jobPosition")
 
-  // val chatMessages by chatViewModel.chatMessages.collectAsState()
+  // text corresponding to interviewer's response
+  val displayText =
+      when {
+        fileData == "Loading interviewer response..." || fileData.isNullOrEmpty() -> {
+          "Processing your audio, please wait..."
+        }
+        else -> "Interviewer's response: $fileData"
+      }
 
+  // rest of UI elements
   Column(
       modifier =
           Modifier.fillMaxSize()
               .padding(AppDimensions.paddingMedium)
               .testTag("RecordingReviewScreen"),
       verticalArrangement = Arrangement.Center,
       horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            text = displayText ?: "",
+            color = MaterialTheme.colorScheme.primary,
+            modifier = Modifier.testTag("ResponseText"))
+
         Button(
             onClick = { player.playFile(audioFile) },
             shape = AppShapes.circleShape,
             modifier = Modifier.testTag("play_button"),
             colors = ButtonDefaults.buttonColors(MaterialTheme.colorScheme.surface),
             contentPadding = PaddingValues(0.dp)) {
-              androidx.compose.material.Icon(
+              Icon(
                   Icons.Outlined.PlayCircleOutline,
                   contentDescription = "Play button",
                   modifier = Modifier.size(30.dp),
```

Comment:
it would be better if  you could make the size dynamic to ensure that it's coherent with the user's screen size

Reply:
I will do that in the offline mode UI PR! Nice catch